src/ansiblelint/utils.py

```diff
@@ -901,3 +898,22 @@ def nested_items(
             yield "list-item", item, parent
             for k, v, p in nested_items(item):
                 yield k, v, p
+
+
+def nested_items_path(
+    data: Union[Dict[Any, Any], List[Any]],
```

Comment:
I guess we can keep it with a docstring that makes it clearer what it is. OTOH, I'm a bit conflicted about it because a variable name should normally reflect its specific semantic meaning within the given context and I felt like it's overly generic: `data=1` and `data=b'some-binary-blob` are also valid examples of the generic context of data, for example.

Reply:
How about using `json` instead of `data because that is really what is supposed to contain, a data structure that is JSON-compatible. Anyway, I am also ok with keeping it data too, the comments are already good-enough.

Reply:
Maybe `node` could be more accurate, I don't know. But I hope you've got the idea of why I think this could be improved :)

Reply:
`collection`?

Reply:
I suppose collection or container match the purpose per https://docs.python.org/dev/library/collections.abc.html.

`data_collection` I guess. Because just `collection` does not set the expectation of its semantics. `"qwerty"` is a collection of characters, right? `{'q', 'w', 'e', 'r', 't', 'y'}` too.

Reply:
done